datafusion/physical-expr/src/equivalence/class.rs

```diff
@@ -546,20 +546,15 @@ impl EquivalenceGroup {
                 .collect::<Vec<_>>();
             (new_class.len() > 1).then_some(EquivalenceClass::new(new_class))
         });
-        // TODO: Convert the algorithm below to a version that uses `HashMap`.
-        //       once `Arc<dyn PhysicalExpr>` can be stored in `HashMap`.
-        // See issue: https://github.com/apache/datafusion/issues/8027
-        let mut new_classes = vec![];
+        let mut new_classes: IndexMap<Arc<dyn PhysicalExpr>, Vec<Arc<dyn PhysicalExpr>>> =
```

Comment:
should we comment what is the key and what is the value in the map? PhyExpr is too abstract imho